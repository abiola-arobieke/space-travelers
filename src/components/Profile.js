import '../styles/Profile.css';
import { useSelector } from 'react-redux';

const Profile = () => {
  const { rocketsItem } = useSelector((store) => store.rockets);

  return (
    <div className="d-flex mt-1">
      <div className="md-6">
        <h3 className="profile-title">Mission</h3>
      </div>
      <div className="md-6 ms-1">
        <h3 className="profile-title">Rockets</h3>
        <ul className="list-container">
          {rocketsItem?.filter((rocket) => rocket.reserved === true).map((reserved) => (
            <li key={reserved.id}>{reserved.name}</li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default Profile;
