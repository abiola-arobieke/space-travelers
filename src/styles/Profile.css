.list-container {
  list-style-type: none;
  margin-block-start: 1em;
  margin-block-end: 1em;
  padding-inline-start: 0;
}

.list-container > li {
  border-left: 1px solid #ccc;
  border-right: 1px solid #ccc;
  border-bottom: 1px solid #ccc;
  font-size: 1rem;
  font-weight: 400;
  padding: 8px 12px 16px;
  text-transform: capitalize;
}

.list-container > li:first-child {
  border-top: 1px solid #ccc;
  border-radius: 2px;
}

.list-container > li:last-child {
  border-radius: 2px;
}

.profile-title {
  font-weight: 500;
}

.ds-flex {
  display: flex;
}

.mid-screen-6 {
  width: 50%;
}

.mar-t-1 {
  margin-top: 1rem;
}

.mar-s-1 {
  margin-left: 1rem;
}
