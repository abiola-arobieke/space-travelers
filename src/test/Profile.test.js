import React from 'react';
import renderer from 'react-test-renderer';
import { useSelector } from 'react-redux';
import Profile from '../components/Profile';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Profile', () => {
  it('renders correctly', () => {
    useSelector.mockImplementation((selectorFn) => selectorFn({
      rockets: {
        rocketsItem: [
          { id: 1, name: 'Rocket 1', reserved: true },
          { id: 2, name: 'Rocket 2', reserved: false },
        ],
      },
      missions: {
        data: [
          { mission_id: '1', mission_name: 'Mission 1', reserved: true },
          { mission_id: '2', mission_name: 'Mission 2', reserved: false },
        ],
      },
    }));

    const component = renderer.create(<Profile />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
