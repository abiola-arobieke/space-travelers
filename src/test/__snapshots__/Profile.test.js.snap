// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<div
  className="d-flex mt-1"
>
  <div
    className="md-6"
  >
    <h3
      className="profile-title"
    >
      Missions
    </h3>
    <ul
      className="list-container"
    >
      <li>
        Mission 1
      </li>
    </ul>
  </div>
  <div
    className="mid-screen-6 mar-s-1"
  >
    <h3
      className="profile-title"
    >
      Rockets
    </h3>
    <ul
      className="list-container"
    >
      <li>
        Rocket 1
      </li>
    </ul>
  </div>
</div>
`;
